### ISSUE TITLE
_CastError Null check operator used on a null value

### ISSUE THREAD

**Issue by daninfpj** [ilteoood/flutter_i18n#193]
I’m sometimes getting this error when calling `translate`.

The problem lies within this function:

```dart
  Map<dynamic, dynamic>? calculateSubmap(final String translationKey) {
    final List<String> translationKeySplitted =
        translationKey.split(this.keySeparator!);
    translationKeySplitted.removeLast();
    Map<dynamic, dynamic>? decodedSubMap = decodedMap;
    translationKeySplitted.forEach((listKey) =>
        decodedSubMap = (decodedSubMap ?? Map())[listKey] ?? Map());
    return decodedSubMap;
  }
```

Specifically, when reaching this line:
```dart
Map<dynamic, dynamic>? decodedSubMap = decodedMap;
```

`decodedMap` is `null`, which makes the function return `null`.

Then, on this line the null check fails:
```dart
final Map<dynamic, dynamic> subMap = calculateSubmap(key)!;
```

Why would `decodedMap` be null at that point? In my app, on a following repaint it is no longer null.

**Comment by ilteoood** [comment]
Can you provide a minimal reproduction example?

**Comment by tomk9** [comment]
Error is when `key` is `""` in `_decodeFromMap`. I paste stacktrace.
![image](https://user-images.githubusercontent.com/15803067/170007238-c48fc082-53c0-48aa-8b8d-7800a07d8f02.png)
![image](https://user-images.githubusercontent.com/15803067/170007302-0f56d0c7-5d6a-4181-b4f2-53566845769e.png)
![image](https://user-images.githubusercontent.com/15803067/170007420-16499557-6618-4168-91c9-f1e322167aad.png)
![image](https://user-images.githubusercontent.com/15803067/170007556-c5fad7a5-b61f-47aa-9862-d801866c13ac.png)
![image](https://user-images.githubusercontent.com/15803067/170007657-b978179e-57eb-4480-8239-f27af962460b.png)

**Comment by tomk9** [comment]
@daninfpj could you test the fix?
```yaml
  flutter_i18n:
    git: https://github.com/ilteoood/flutter_i18n
```
It helped me. Maybe after @daninfpj confirmation, @ilteoood could release a new version on pub and close the issue?

**Comment by ilteoood** [comment]
Hi guys, I'm in honeymoon right now.
The suggested workaround could be used until I come back, at the end of this week.

**Comment by ilteoood** [comment]
A fix should have been published with version 0.32.3
